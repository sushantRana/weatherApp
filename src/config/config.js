module.exports = {
  APPID: 'a23c219b61541dabd9bd6b7bb9d7ee0f',
  GoogleURL: 'https://maps.googleapis.com/maps/api/js?key=AIzaSyBP2_Okzm1JzT2T1IJNOswN2W1OXnOb77E&v=3.exp&libraries=geometry,drawing,places',
  weatherAPIURL: 'http://api.openweathermap.org/data/2.5/forecast?'
};
