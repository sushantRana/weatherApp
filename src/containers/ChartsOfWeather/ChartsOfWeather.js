import React from 'react';
import axios from 'axios';
import { Line } from 'react-chartjs-2';
import { weatherAPIURL, APPID } from '../../config/config';
import styles from './ChartsOfWeather.css'

const toCelcius = value => (value - 273).toPrecision(2);
const dataTemplate = {
  labels: [''],
  datasets: [
    {
      label: 'Temperature',
      backgroundColor: 'rgba(255,99,132,0.2)',
      borderColor: 'rgba(255,99,132,1)',
      borderWidth: 1,
      hoverBackgroundColor: 'rgba(255,99,132,0.4)',
      hoverBorderColor: 'rgba(255,99,132,1)',
      data: [0]
    }
  ]
};
class ChartsOfWeather extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      selectedData : false
    }
  }
  formatData = (rawdata) => {
    const labelsData = rawdata.map((current) => toCelcius(current.main.temp));
    const labels = rawdata.map((current) => current.dt_txt);

    const formattedData = { ...dataTemplate };
    formattedData.labels = labels;
    formattedData.datasets[0].data = labelsData;
    return formattedData;
  }
  componentDidMount() {
    const { lat, lng } = this.props;
    axios.get(`${weatherAPIURL}lat=${lat}&lon=${lng}&appid=${APPID}`)
      .then(res => {
        this.setState({
          selectedData: this.formatData(res.data.list)
        })
      })
  }
  
  render() {
        return (
          <div className={styles.mainDiv}>
            <Line
              data={this.state.selectedData}
              width={300}
              height={400}
              options={{
                maintainAspectRatio: false
              }}
            />
          </div>
        )
  }
}
export default ChartsOfWeather;