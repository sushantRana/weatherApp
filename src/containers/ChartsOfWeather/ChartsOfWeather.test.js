import React from 'react';
import { shallow } from 'enzyme';
import axios from 'axios';
import ChartsOfWeather from './ChartsOfWeather';
import { weatherAPIURL, APPID } from '../../config/config';

jest.mock('axios', () => {
  const receivedData = {
    data: {"list":[{"dt":1531310400,"main":{"temp":301.41,"temp_min":298.633,"temp_max":301.41,"pressure":1013.56,"sea_level":1014.27,"grnd_level":1013.56,"humidity":100,"temp_kf":2.77},"weather":[{"id":501,"main":"Rain","description":"moderate rain","icon":"10d"}],"clouds":{"all":92},"wind":{"speed":8.62,"deg":247.501},"rain":{"3h":6.605},"sys":{"pod":"d"},"dt_txt":"2018-07-11 12:00:00"}]}
  };
  
  return {
    get: jest.fn(() => Promise.resolve(receivedData)),
  };
});
const props = {
	lat: 19.0759837,
	lng: 72.8776559
}
const lat= 19.0759837;
const lng= 72.8776559;
const charts = shallow(<ChartsOfWeather {...props}/>);
it('fetch data on #componentDidMount of ChartsOfWeather', () => {
	
  const url = `${weatherAPIURL}lat=${lat}&lon=${lng}&appid=${APPID}`;
  charts.instance().componentDidMount();
  expect(axios.get).toHaveBeenCalled();
  expect(axios.get).toHaveBeenCalledWith(url);

});
it ('renders without crashing', () => {
    expect(charts).toMatchSnapshot();
})