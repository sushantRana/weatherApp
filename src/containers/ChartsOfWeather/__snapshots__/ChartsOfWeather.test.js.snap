// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ChartsOfWeather
    lat={19.0759837}
    lng={72.8776559}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Line
        data={false}
        height={400}
        options={
                Object {
                        "maintainAspectRatio": false,
                      }
        }
        width={300}
/>,
      "className": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "data": false,
        "height": 400,
        "options": Object {
          "maintainAspectRatio": false,
        },
        "width": 300,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Line
          data={false}
          height={400}
          options={
                    Object {
                              "maintainAspectRatio": false,
                            }
          }
          width={300}
/>,
        "className": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "data": false,
          "height": 400,
          "options": Object {
            "maintainAspectRatio": false,
          },
          "width": 300,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
